capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java

```diff
@@ -0,0 +1,129 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+
+import com.google.sps.data.Club;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that allows users to post and get comments. */
+@WebServlet("/api/clubs")
+public class ClubServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String stringID = request.getParameter("id");
+    Query query = new Query("Club");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    long id = Long.parseLong(stringID);
+
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<Entity> entities = results.asList(FetchOptions.Builder.withLimit(5));
+    List<Club> clubs = new ArrayList<>();
+    for(Entity e : entities) {
+      String name = (String) e.getProperty("name");
+      String description = (String) e.getProperty("description");
+      String announcement = (String) e.getProperty("announcement");
+
+      @SuppressWarnings("unchecked") // Cast can't verify generic type.
+      Collection<String> posts = (ArrayList<String>) e.getProperty("posts");
+
+      @SuppressWarnings("unchecked") // Cast can't verify generic type.
+      Collection<Long> memberIDs = (HashSet<Long>) e.getProperty("memberIDs");
+      
+      @SuppressWarnings("unchecked") // Cast can't verify generic type.
+      Collection<Long> inviteIDs = (HashSet<Long>) e.getProperty("inviteIDs");
+
+      long ownerID = (long) e.getProperty("ownerID");
+      String gbookID = (String) e.getProperty("gbookID");
+      Club club = new Club(id, name, description, announcement, ownerID, gbookID);
```

Comment:
Isn't the `id` the same for all of the returned clubs here (the `id` from `request.getParameter("id")` above)? Is this intentional (doesn't each entity `e` have its own `id`, `e.getProperty('id')`)?

Reply:
Meant to remove the line getting id from the request, as we were having trouble filtering in Datastore. For now our doGet should grab all clubs. I changed it to grab ID from ```e.getProperty("id"); ```